src/node_report_module.cc

```diff
@@ -129,12 +130,14 @@ static void SetSignal(const FunctionCallbackInfo<Value>& info) {
 }
 
 static void ShouldReportOnFatalError(const FunctionCallbackInfo<Value>& info) {
+  Mutex::ScopedLock lock(node::per_process::cli_options_mutex);
```

Comment:
will defining `report_on_fatalerror` as `std::atomic<bool>` be a better approach?

Reply:
@gireeshpunathil You could do that, but I doubt that it’s worth the extra complexity.